install/kubernetes/helm/istio/charts/gateways/templates/service.yaml

```diff
@@ -27,6 +27,15 @@ spec:
     {{ $key }}: {{ $val }}
     {{- end }}
   ports:
+{{- if $spec.meshExpansion }}
+    - port: 15005
+      targetPort: 15005
+      name: tcp-pilot
+    - port: 8060
+      targetPort: 8060
+      name: tcp-citadel
+{{- end }}
```

Comment:
Do we need to overload the gateway for this ?

Reply:
Since this works only with ilb why not have a separate ilb gateway instead of overloading consumer facing ingress gateway? Also this is not the right place to add ports. It’s in values.yaml

Reply:
On the first question: somehow the gateway needs to listen on the port. We can add it in values.yaml, but it's harder for the user, we may have other boiler plage.

On the second: this is not based on ILB. We may add an ILB if anyone asks, but current solution is exposing mtls only and can be used on the regular gateway

Having a separate gateway is possible - that's why it's an option on the gateway, a user can add add more gateways and select to enable mesh expansion ports selectively.